src/routes/github/subscription-deferred-install/github-subscription-deferred-install-router.ts

```diff
@@ -0,0 +1,73 @@
+import { NextFunction, Request, Response, Router } from "express";
+import { extractSubscriptionDeferredInstallPayload, SubscriptionDeferredInstallPayload } from "services/subscription-deferred-install-service";
+import { Installation } from "models/installation";
+import { GithubServerAppMiddleware } from "middleware/github-server-app-middleware";
+import { GithubAuthMiddleware } from "routes/github/github-oauth";
+import {
+	GithubSubscriptionDeferredInstallGet
+} from "routes/github/subscription-deferred-install/github-subscription-deferred-install-get";
+import { booleanFlag, BooleanFlags } from "config/feature-flags";
+import {
+	GithubSubscriptionDeferredInstallPost
+} from "routes/github/subscription-deferred-install/github-subscription-deferred-install-post";
+
+const GithubSubscriptionDeferredInstallRouter = Router({ mergeParams: true });
+
+const subRouter = Router({ mergeParams: true });
+
+const INVALID_PAYLOAD_ERR = "Invalid payload";
+
+GithubSubscriptionDeferredInstallRouter.use("/request/:requestId", subRouter);
+
+const validatePayloadAndPopulateJiraHost = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+	let parsedPayload: SubscriptionDeferredInstallPayload | undefined;
+	try {
+		parsedPayload = await extractSubscriptionDeferredInstallPayload(req);
+	} catch (err) {
+		req.log.warn({ err }, "Cannot deserialize");
+		res.status(400).json({ error: INVALID_PAYLOAD_ERR });
+		return;
+	}
+
+	const installation = await Installation.findByPk(parsedPayload.installationIdPk);
+	if (!installation) {
+		req.log.warn("No installation");
+		res.status(400).json({ error: INVALID_PAYLOAD_ERR });
+		return;
+	}
+
+	if (!await booleanFlag(BooleanFlags.ENABLE_SUBSCRIPTION_DEFERRED_INSTALL, installation.jiraHost)) {
+		res.status(401).json({ error: "Feature is disabled" });
+		return;
+	}
+
+	res.locals.jiraHost = installation.jiraHost;
```

Comment:
/nit
**IMPORTANT**
So I am a bit concern about the security model here.

ps: lol, nit and important at the same time.

I understand this url with the `requestId` is supposed to be shortLive, however we don't have a timer on the requestId in redis? Or do we? And if any malicious users intercept the `requestId` somehow (Since the jiraAdmin will send this url to gitHubAdmin somehow, the requestId is in the url path), that malicious user can 
1. Hit the url with with its own `gitHubToken`, so can connect his own github org to the victims' jiraHost?
2. Do we have a timer on the requestId on redis the auto destroy?

Reply:
I agree with Gary here (although maybe disagree with the nit part 😜... where's :wink_: when ya need it?).

To Gary's first point, there's definitely potential for someone to intercept the requestId, hitting the URL with their own GitHub token, and connecting their GitHub organization to the someone else's Jira host. We definitely want to make sure we have some kind of expiration mechanism for the requestId. And to his second point: can we check if we do have an expiration mechanism in place? If we don't we definitely want to set a TTL on the requestId key in Redis to destroy expired requests.

Reply:
We setup TTL in redis when we put the record, don't we?

1. No, they cannot. The user must be an admin of the github org that is being connected.
2. yes, we have a timer REDIS_CLEANUP_TIMEOUT, currently set up to 2 days
3 . (to Rachelle), they cannot connect to their Jira host because it is inside the payload that they cannot change

<img width="857" alt="image" src="https://github.com/atlassian/github-for-jira/assets/20631664/0d085113-b33d-44ff-8ca1-9c56029debdf">

Reply:
Maybe to clarify it a little bit more:

1. admin prepares a request for connection. Noone can prepare this request for that Jira except of the admin, cause the payload is stored on the backend. Even if requestId is known, that's nothing, just an admin wanting to connect an org to a Jira...
2. to finish the connection, the user must know requestId and to be the GitHub admin of the org that is being requested to connect. Again, if you are a GitHub admin of some other org, or not a GitHub admin at all, you won't be able even to read the details of the request

Reply:
feel free to ping me in Slack to discuss if you could see any security threat that I cannot